lib/npg_pipeline/function/s3_archiver.pm

```diff
@@ -0,0 +1,265 @@
+package npg_pipeline::function::s3_archiver;
+
```

Comment:
The new functions have to be registered in npg_pipeline::pluggable::registry, which provides mapping from names in input function graph to modules and methods